### PR TITLE
Adds Filth Tolerant trait

### PR DESCRIPTION
**Pull request by DangerNoodle**
Adds trait to mitigate filthy clothing effect. Reasoning: psychopath is a trait that affects morale-only things, and flithy clothes are currently a morale effect. Trait point cost is inverse of squeamish, can be increased if desired.

Practical effects for filthy clothes will still apply if they exist or are added.

### CODE REVIEW

**Comment by DangerNoodle** [comment]
?

Not very used to this.

**Comment by Treah** [comment]
Should make it so that they actively enjoy being filthy and will get a hit in moral if they are clean.

**Comment by DangerNoodle** [comment]
Would be odd. Working on fixing code in general.

**Comment by DangerNoodle** [comment]
```
astyle regressions found.
Formatted  src/morale.cpp
make: *** [astyle-check] Error 1
make: INTERNAL: Exiting with 3 jobserver tokens available; should be 2!
```

This is new. I do not know how I styled it wrong. Surprised this is the only error reported in second commit.

**Comment by Zilenan91** [comment]
This is just a band-aid fix to a problem that should just be removed.

If people don't like filthiness they'll just be told to take the trait rather than there being actual debate on if it's worth keeping it in the game in the first place

**Comment by DangerNoodle** [comment]
I agree, but the feature likely will not be removed. It is better to have some way around this trait, than to leave it simultaneously annoying and mandatory.

Now, what am I missing here? I seem to have fixed the actual code, but some part of my formatting is generating an error. I'm not very familiar with the astyle demands.

File: src/morale.cpp [hunk 1]
```diff
@@ -299,6 +300,7 @@ player_morale::player_morale() :
     const auto set_badtemper      = std::bind( &player_morale::set_permanent, _1, MORALE_PERM_BADTEMPER,
                                     _2, nullptr );
     const auto set_stylish        = std::bind( &player_morale::set_stylish, _1, _2 );
+ const auto set_filth_tolerant = std::bind( &player_morale::set_filth_tolerant, _1, _2 );
```
Comment (BorkBorkGoesTheCode) [inline src/morale.cpp]: Is this it?

Comment (DangerNoodle) [inline src/morale.cpp]: Outdated. Fixing solved nothing.

**Comment by Zireael07** [comment]
To fix the style part you need to find astyle program and run it. @Coolthulhu probably knows more than I :P

**Comment by DangerNoodle** [comment]
I have no idea where to look for this, but I will try.

**Comment by DangerNoodle** [comment]
Still working on procuring astyle checker, updating repository, and trying it. But the only visible possible style failure I can see might be caused by `FILTH_TOLERANT` being just long enough to affect spacing for `player_morale::mutation_data::clear()`

**Comment by DangerNoodle** [comment]
There. Just a matter of getting astyle instead of relying upon my poor memory of style rules.

Now I will just need to make Squeamish and Filth Tolerance cancel each other out. In addition, should I add the trait to the medieval peasant profession, to add to the intended humorous flavor?

**Comment by Coolthulhu** [comment]
I'd much rather get rid of the whole filthiness morale thing.
Quick fixes like that make it look like we're OK with it.

**Comment by DangerNoodle** [comment]
But is that ever going to happen? We can always remove this trait if we remove filth morale entirely.

Dropping this pull request because of that leaves the current system broken, and still forces it upon players.

**Comment by mugling** [comment]
Superseded by #18233

**Comment by DangerNoodle** [comment]
Ah, thank you. Apologies for not being on to close this myself.

**Pull request closed without merging**